Gems/GradientSignal/Code/Include/GradientSignal/Components/SurfaceAltitudeGradientComponent.h

```diff
@@ -105,7 +107,18 @@ namespace GradientSignal
         void AddTag(AZStd::string tag) override;
 
     private:
-        mutable AZStd::recursive_mutex m_cacheMutex;
+        static float CalculateAltitudeRatio(const SurfaceData::SurfacePointList& points, float altitudeMin, float altitudeMax)
```

Comment:
I don't actually know why only the first point is used, it looks like a bug to me. I wasn't trying to change the logic on any of these, but in this case, it would make sense to improve it.  I'll fix it to use the height value from any of the points.